### PR TITLE
Use localStorage instead of cookies if available

### PR DESCRIPTION
**Pull request by WMDE-Fisch**
Bug: T143777

### CODE REVIEW

File: DeepCat.js [hunk 1]
```diff
@@ -500,7 +500,33 @@
 	}
 
 	function hasHintCookie() {
-		return mw.cookie.get( '-deepcat-hintboxshown' ) === makeHintboxCookieToken( mw.msg( 'deepcat-hintbox-text' ) );
+		var storedData;
+		if( typeof Storage !== 'undefined' ) {
```
Comment (thiemowmde) [inline DeepCat.js:504]: Just curious, is it good practice to check this instead of checking for `localStorage`?

**Comment by manicki** [comment]
I was thinking the same thing as @thiemowmde but apparently there is the way other people on internets use.

I also think those functions involved here could be renamed so they don't contain the "cookie" word, as cookie is now only a legacy browser fallback.

Few more remarks:
- if the user already has "no hints" cookie it will be ignored and she will get hints again (once closed now, the value in local storage is added and the hint will then never appear again - for a month). But this might not be that bad (if you close the hint immediately you will only see it this single time)
- cookies seems/seemed to be wiki-specific (e.g. `dewiki-deepcat-hintboxshown`) whereas local storage only uses the item called `mw-deepcat-hintboxshown` which in my understanding means it will disable hints in all wikis. Am I right or am I missing something here? And if it is as I say, this might or might not be intended.

But those two could be addressed in follow ups (if required).

**Comment by WMDE-Fisch** [comment]
@wmde-manicki Thanks for your notes on this.

> if the user already has "no hints" cookie it will be ignored and she will get hints again (once closed now, the value in local storage is added and the hint will then never appear again - for a month). But this might not be that bad (if you close the hint immediately you will only see it this single time)

 I thought about adding a check to delete existing cookies. But since they will be gone after 4 weeks anyway it might be to much of a hassle.And the code for that will become useless then.

> cookies seems/seemed to be wiki-specific (e.g. dewiki-deepcat-hintboxshown) whereas local storage only uses the item called mw-deepcat-hintboxshown which in my understanding means it will disable hints in all wikis. Am I right or am I missing something here? And if it is as I say, this might or might not be intended.

I thought that would be an improvement. :-)

**Comment by manicki** [comment]
> I thought about adding a check to delete existing cookies. But since they will be gone after 4 weeks anyway it might be to much of a hassle.And the code for that will become useless then.

Agreed

> I thought that would be an improvement. :-)

I think so too. This is just changing the previous behaviour, so I thought I'll complain about it :) I believe it is OK to do so unless it specifically required to have wiki-specific hint settings.

**Pull request merged**